### PR TITLE
Provide more informative kernel names in Inductor

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96170
* __->__ #95940

Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/fx/proxy.py [hunk 1]
```diff
@@ -164,19 +164,15 @@ def create_proxy(self, kind: str, target: Target, args: Tuple[Any, ...], kwargs:
         if fx_traceback.has_preserved_node_meta():
             current_meta: Dict[str, Any] = fx_traceback.get_current_meta()
 
-            # Explicitly set the stack_trace, nn_module_stack and source_fn on the node.meta
-            # If other meta fields are needed, they can be added here
             stack_trace = current_meta.get("stack_trace")
```
Comment (SherlockNoMad) [inline torch/fx/proxy.py:167]: nit, stack_trace can also be included in the copy_meta_fields

File: torch/fx/experimental/proxy_tensor.py [hunk 2]
```diff
@@ -471,6 +472,23 @@ def wrapped(*proxies):
 
     return wrapped
 
+ORIGINAL_ATEN = None
+@contextmanager
+def set_original_aten_op(func):
+    global ORIGINAL_ATEN
+    if ORIGINAL_ATEN is None:
+        ORIGINAL_ATEN = func
+        fx_traceback.current_meta['original_aten'] = func
```
Comment (SherlockNoMad) [inline torch/fx/experimental/proxy_tensor.py:481]: condition on 
fx_traceback.has_preserved_node_meta() is True

**Comment by SherlockNoMad** [comment]
Need some test. see
test_transformer_preserves_nn_module_stack_for_get_attr()  in  test_fx.py 
test_export_with_stack_trace() in test_export.py

File: torch/_dynamo/output_graph.py [hunk 3]
```diff
@@ -810,6 +810,9 @@ def create_proxy(
 
         if kind in {"call_function", "call_method"}:
             rv.node.meta["source_fn"] = target
+        elif kind == "call_module":
+            # For modules we store the class
+            rv.node.meta["source_fn"] = rv.node.meta["nn_module_stack"][target][1]
```
Comment (SherlockNoMad) [inline torch/_dynamo/output_graph.py:815]: cc @jerryzh168

**Review by SherlockNoMad — APPROVED** [review]
per comment

**Review by ngimel — APPROVED** [review]

**Review by jansel — APPROVED** [review]

File: torch/fx/proxy.py (hunk 1 above)
Comment (Chillee) [reply torch/fx/proxy.py]: Stack trace is stored as an attribute directly on node, while the other metadata are stored in the meta field.

**Pull request closed without merging**